src/Haskell/Ide/Engine/Plugin/Hoogle.hs

```diff
@@ -152,7 +152,19 @@ renderTarget t = T.intercalate "\n" $
 -- If an error occurs, such as no hoogle database has been found,
 -- or the search term has no match, an empty list will be returned.
 searchModules :: T.Text -> IdeM [T.Text]
-searchModules = fmap (nub . take 5) . searchTargets (fmap (T.pack . fst) . targetModule)
+searchModules = fmap (map fst) . searchModules'
+
+-- | Just like 'searchModules', but includes the signature of the search term
+-- that has been found in the module.
+searchModules' :: T.Text -> IdeM [(T.Text, T.Text)]
+searchModules' = fmap (nub . take 5)
+  . searchTargets
+    (\target
+     -> (\modTarget -> (T.pack $ fst modTarget, normaliseItem . T.pack $ targetItem target))
```

Comment:
can we maybe take this lambda out into the `where` clause as a 2 argument function? I think it would make it easier to read

Reply:
you are right